### PR TITLE
[MNO-287] Developer Section in My Account

### PR DESCRIPTION
**Pull request by hedudelgado**
@x4d3 @alexnoox  @ouranos

### CODE REVIEW

**Comment by hedudelgado** [comment]
https://github.com/maestrano/mno-enterprise/pull/153

**Comment by hedudelgado** [comment]
https://github.com/maestrano/maestrano-hub/pull/244

File: src/app/views/account/account.html [hunk 1]
```diff
@@ -92,52 +92,54 @@ <h2 class="section-title">{{ 'mno_enterprise.templates.dashboard.account.my_acco
...
-        <div class="row">
-          <div class="col-sm-3 field-name">{{ 'mno_enterprise.templates.dashboard.account.secret_key' | translate }}:</div>
-          <div class="col-sm-9 field-name">
-            <span ng-show="vm.user.revealedSecretKey">{{vm.user.model.api_secret}}</span>
-            <button ng-hide="vm.user.revealedSecretKey" class='btn btn-warning' ng-click="vm.user.revealSecretKey()">{{ 'mno_enterprise.templates.dashboard.account.reveal' | translate }}</button>
-            <button ng-show="vm.user.revealedSecretKey" class='btn btn-warning' ng-click="vm.user.revealSecretKey()">{{ 'mno_enterprise.templates.dashboard.account.conceal' | translate }}</button>
+        <div ng-if="!vm.user.model.api_key || !vm.user.model.api_secret" class="row">
+          <div class='col-md-12'>
+            <p>{{ 'mno_enterprise.templates.dashboard.account.are_you_developer' | translate }}</p>
+            <button class='btn btn-warning' ng-click="vm.user.registerDeveloper()">{{ 'mno_enterprise.templates.dashboard.account.register!' | translate }}</button>
```
Comment (alexnoox) [inline src/app/views/account/account.html]: Avoid to use special characters (!) in keys names.

File: src/app/views/account/account.html [hunk 2]
```diff
@@ -92,52 +92,54 @@ <h2 class="section-title">{{ 'mno_enterprise.templates.dashboard.account.my_acco
...
-              <tbody>
-                <tr ng-repeat="org in vm.user.model.organizations track by $index">
-                  <td>{{org.name}}</td>
-                  <td>{{org.uid}}</td>
-                </tr>
-              </tbody>
-            </table>
+        <div ng-if="vm.user.model.api_key && vm.user.model.api_secret">
+          <div class="row">
+            <div class="col-sm-3 field-name">{{ 'mno_enterprise.templates.dashboard.account.api_key' | translate }}: </div>
```
Comment (alexnoox) [inline src/app/views/account/account.html]: Is the class ".field-name" defined in mnoe-angular?

File: src/app/views/account/account.html [hunk 3]
```diff
@@ -92,52 +92,54 @@ <h2 class="section-title">{{ 'mno_enterprise.templates.dashboard.account.my_acco
...
+      <div uib-accordion-group is-open="isDevOpen" id="myaccount-dev">
+        <div uib-accordion-heading>{{ 'mno_enterprise.templates.dashboard.account.developer' | translate }} <span class="fa pull-right" ng-class="{'fa-caret-down': !isDevOpen, 'fa-caret-up': isDevOpen}"></span>
         </div>
-        <div class="row">
-          <div class="col-sm-3 field-name">{{ 'mno_enterprise.templates.dashboard.account.secret_key' | translate }}:</div>
-          <div class="col-sm-9 field-name">
-            <span ng-show="vm.user.revealedSecretKey">{{vm.user.model.api_secret}}</span>
-            <button ng-hide="vm.user.revealedSecretKey" class='btn btn-warning' ng-click="vm.user.revealSecretKey()">{{ 'mno_enterprise.templates.dashboard.account.reveal' | translate }}</button>
-            <button ng-show="vm.user.revealedSecretKey" class='btn btn-warning' ng-click="vm.user.revealSecretKey()">{{ 'mno_enterprise.templates.dashboard.account.conceal' | translate }}</button>
+        <div ng-if="!vm.user.model.api_key || !vm.user.model.api_secret" class="row">
```
Comment (alexnoox) [inline src/app/views/account/account.html]: Shouldn't it be a `&&`?

File: src/app/views/account/account.controller.coffee [hunk 4]
```diff
@@ -90,6 +90,20 @@ angular.module 'mnoEnterpriseAngular'
       vm.isLocalizationOpen = false
 
       # ----------------------------------------------------
+      # Developer Section
+      # ----------------------------------------------------
+      vm.isDevOpen = false
+      vm.user.registerDeveloper = ->
+        MnoeCurrentUser.registerDeveloper(vm.user.model).then(
```
Comment (alexnoox) [inline src/app/views/account/account.controller.coffee]: Why do you pass an object? `MnoeCurrentUser.registerDeveloper()` doesn't have a param in its signature.

File: src/app/views/account/account.controller.coffee [hunk 5]
```diff
@@ -90,6 +90,20 @@ angular.module 'mnoEnterpriseAngular'
...
+          (success) ->
+            angular.extend(vm.user.model, success)
+        )
+        
+      vm.user.revealedSecretKey = false
+      vm.user.revealSecretKey = ->
```
Comment (alexnoox) [inline src/app/views/account/account.controller.coffee]: Rename it `switchSecretKey`.

File: src/app/views/account/account.html [hunk 6]
```diff
@@ -90,6 +90,58 @@ <h2 class="section-title">{{ 'mno_enterprise.templates.dashboard.account.my_acco
     </div>
 
     <!--
+      Developer section
+    -->
+    <div ng-if='vm.isDeveloperSectionEnabled'>
+      <div uib-accordion-group is-open="isDevOpen" id="myaccount-dev">
```
Comment (alexnoox) [inline src/app/views/account/account.html]: Is the id `myaccount-dev` used?

Comment (alexnoox) [inline src/app/views/account/account.html]: shouldn't it be `vm.isDevOpen`?

File: src/app/views/account/account.html [hunk 7]
```diff
@@ -90,6 +90,58 @@ <h2 class="section-title">{{ 'mno_enterprise.templates.dashboard.account.my_acco
...
+        <div uib-accordion-heading>{{ 'mno_enterprise.templates.dashboard.account.developer' | translate }} <span class="fa pull-right" ng-class="{'fa-caret-down': !isDevOpen, 'fa-caret-up': isDevOpen}"></span>
```
Comment (alexnoox) [inline src/app/views/account/account.html]: Same as above

File: src/app/views/account/account.controller.coffee [hunk 8]
```diff
@@ -90,6 +90,20 @@ angular.module 'mnoEnterpriseAngular'
...
+      # Developer Section
+      # ----------------------------------------------------
+      vm.isDevOpen = false
+      vm.user.registerDeveloper = ->
+        MnoeCurrentUser.registerDeveloper(vm.user.model).then(
+          (success) ->
+            angular.extend(vm.user.model, success)
+        )
+        
+      vm.user.revealedSecretKey = false
```
Comment (alexnoox) [inline src/app/views/account/account.controller.coffee]: Rename it `vm.isSecretKeyRevealed`

File: src/app/views/account/account.less [hunk 9]
```diff
@@ -2,4 +2,9 @@
   .row:not(:first-child) {
     margin-top: 15px;
   }
+
+  .col-sm-3 {
+    width: 25%;
+    font-weight: bold;
+  }
```
Comment (alexnoox) [inline src/app/views/account/account.less]: Can you find another way to do this?

File: src/app/views/account/account.html [hunk 10]
```diff
@@ -90,6 +90,58 @@ <h2 class="section-title">{{ 'mno_enterprise.templates.dashboard.account.my_acco
...
+          <div class="row">
+            <div class="col-sm-3 field-name">{{ 'mno_enterprise.templates.dashboard.account.api_key' | translate }}: </div>
+            <div class="col-sm-9 field-name">{{vm.user.model.api_key}}</div>
+          </div>
+          <div class="row">
+            <div class="col-sm-3 field-name">{{ 'mno_enterprise.templates.dashboard.account.secret_key' | translate }}:</div>
+            <div class="col-sm-9 field-name">
+              <span ng-show="vm.user.revealedSecretKey">{{vm.user.model.api_secret}}</span>
+              <button ng-hide="vm.user.revealedSecretKey" class='btn btn-warning' ng-click="vm.user.revealSecretKey()">{{ 'mno_enterprise.templates.dashboard.account.reveal' | translate }}</button>
+              <button ng-show="vm.user.revealedSecretKey" class='btn btn-warning' ng-click="vm.user.revealSecretKey()">{{ 'mno_enterprise.templates.dashboard.account.conceal' | translate }}</button>
```
Comment (alexnoox) [inline src/app/views/account/account.html]: Rename the buttons `show`/`hide`

**Comment by hedudelgado** [comment]
@alexnoox I have added the changes you requested.
Thanks a lot.

**Pull request merged**